bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableVaneerSettingsFactory.java

```diff
@@ -0,0 +1,312 @@
+ */
+public class BigtableVaneerSettingsFactory {
+
+  /** Constant <code>LOG</code> */
+  private static final Logger LOG = new Logger(BigtableVaneerSettingsFactory.class);
+
+  /**
+   * To create an instance of {@link BigtableDataSettings} from {@link BigtableOptions}.
+   *
+   * @param options a {@link BigtableOptions} object.
+   * @return a {@link BigtableDataSettings} object.
+   * @throws IOException if any.
+   */
+  public static BigtableDataSettings createBigtableDataSettings(final BigtableOptions options) throws IOException {
+    checkState(options.getProjectId() != null, "Project ID is required");
+    checkState(options.getInstanceId() != null, "Instance ID is required");
+    checkState(options.getRetryOptions().enableRetries(), "Disabling retries is not currently supported.");
+
+    BigtableDataSettings.Builder builder = BigtableDataSettings.newBuilder();
+    String endpoint = options.getDataHost() + ":" + options.getPort();
+
+    InstanceName instanceName = InstanceName.newBuilder().setProject(options.getProjectId())
+        .setInstance(options.getInstanceId()).build();
+    builder.setInstanceName(instanceName);
+    builder.setAppProfileId(options.getAppProfileId());
+
+    builder.setEndpoint(endpoint);
+
+    builder.setCredentialsProvider(buildCredentialProvider(options.getCredentialOptions()));
+
+    buildBulkMutationsSettings(builder, options);
+
+    buildCheckAndMutateRowSettings(builder, options.getCallOptionsConfig().getShortRpcTimeoutMs());
+
+    buildReadModifyWriteSettings(builder, options.getCallOptionsConfig().getShortRpcTimeoutMs());
+
+    buildReadRowsSettings(builder, options);
+
+    buildMutateRowSettings(builder, options);
+
+    buildSampleRowKeysSettings(builder, options);
+
+    String userAgent = BigtableVersionInfo.CORE_UESR_AGENT + "," + options.getUserAgent();
+    HeaderProvider headers = FixedHeaderProvider.create(GrpcUtil.USER_AGENT_KEY.name(), userAgent);
+
+    builder.setTransportChannelProvider(
+        InstantiatingGrpcChannelProvider.newBuilder()
+            .setHeaderProvider(headers)
+            .setEndpoint(endpoint)
+            .setPoolSize(options.getChannelCount())
+            .build());
+
+    return builder.build();
+  }
+
+  /**
+   * To create an instance of {@link BigtableTableAdminSettings} from {@link BigtableOptions}.
+   *
+   * @param options a {@link BigtableOptions} object.
+   * @return a {@link BigtableTableAdminSettings} object.
+   * @throws IOException if any.
+   */
+  public static BigtableTableAdminSettings createTableAdminSettings(BigtableOptions options)
+      throws IOException {
+    checkState(options.getProjectId() != null, "Project ID is required");
+    checkState(options.getInstanceId() != null, "Instance ID is required");
+
+    BigtableTableAdminSettings.Builder adminBuilder = BigtableTableAdminSettings.newBuilder();
+
+    adminBuilder.setInstanceName(com.google.bigtable.admin.v2.InstanceName
```

Comment:
Please add a note to stop using typesafe names

Reply:
Updated classes to use with raw ids